### ISSUE TITLE
Brian - MCTS

### ISSUE THREAD

**Issue by subtotechnoblade** [subtotechnoblade/Grok_Alpha_Zero#9]
Get the monte carlo tree search working with a dummy policy and value

Store child statistics in parent node
Save as much memory as possible 

methods - _puct_select(node) -> node (child):
if the parent.visits < len(children) -> child with no visits will equal len(children)
return that child

else calculate PUCT with a numba function and return the index in which puct is the highest (argmax(PUCT))
given by PUCT_score = (cum_value / child_visits) + c * prob_prior * (sqrt(parent_visits) / (child_visits + 1))

_expand(node) -> None:
perform a inference call to the neural network to get the policy and value for that node
fills in that node with the prob_prior <- policy and value += -value # value is negated because of player POV

_backprop(node, value)

while node.parent is not None:
node.parent.value[node.index] += value# because the node's stats are stored in its parent
node.parent.visits[node.index] += 1
value *= -1
node = node.parent

calculate the probability distribution with softmax or perhaps just with normalization
return the argmax of the probs and the move associated with it
